framework/src/org/apache/cordova/CordovaPlugin.java

```diff
@@ -359,4 +361,38 @@ public boolean onReceivedClientCertRequest(CordovaWebView view, ICordovaClientCe
      */
     public void onConfigurationChanged(Configuration newConfig) {
     }
+
+    /**
+     * Called by the Plugin Manager when we need to actually request permissions
+     *
+     * @return              Returns the permission that was stored in the plugin
+     */
+
+    public String[] getPermissionRequest() {
+        return permissions;
+    }
+
+    public boolean hasPermisssion() {
+        for(String p : permissions)
+        {
+            if(PackageManager.PERMISSION_DENIED == cordova.getActivity().checkSelfPermission(p))
```

Comment:
That's probably the case.  I'm almost tempted to remove the code from this method entirely, since this really exists only for a Geolocation plugin.  The geolocation plugin is a special case, and no other plugins work this way.